### PR TITLE
Add RawContext() method to be used by children contexts (AO-17292)

### PR DESCRIPTION
**Pull request by adamiklukasz**
https://swicloud.atlassian.net/browse/AO-17292

Required to simplify kubernetes client calls.

Want to:
```
	fetchCtx, cancelFn := context.WithCancel(ctx.RawContext())
	defer cancelFn()

	resp, err := c.fetch(fetchCtx)
	if err != nil {
		return fmt.Errorf("collect error: %v", err)
	}
```

Instead of:
```
	fetchCtx, cancelFn := context.WithCancel(context.Background())

	doneCh := make(chan struct{})

	go func() {
		select {
		case <-ctx.Done():
			cancelFn()
		case <-doneCh:
		}
	}()

	resp, err := c.fetch(fetchCtx)
	if err != nil {
		return fmt.Errorf("collect error: %v", err)
	}

	select {
	case <-doneCh:
	default:
	}
```

Example:
https://github.com/librato/snap-plugin-collector-kubernetes/blob/ao-17292-v2/kubernetes/collector/collector.go#L31

### CODE REVIEW

**Review by mpach-swi — DISMISSED** [review]

File: v2/internal/plugins/common/proxy/context.go [hunk 1]
```diff
@@ -172,6 +172,13 @@ func (c *Context) ResetWarnings() {
 	c.sessionWarnings = []types.Warning{}
 }
 
+func (c *Context) RawContext() context.Context {
+	c.ctxMu.RLock()
```
Comment (lootek) [inline v2/internal/plugins/common/proxy/context.go:176]: what's the point of locking since the internal ref is returned anyway?

**Review by lootek — APPROVED** [review]

**Review by lootek — COMMENTED** [review]
could you please link to the relevant link in the k8s collector?

File: v2/internal/plugins/common/proxy/context.go (hunk 1 above)
Comment (lootek) [reply v2/internal/plugins/common/proxy/context.go:176]: and what would happen when the c.ctx changes? the code that fetched the old one is gonna hang?

Comment (adamiklukasz) [reply v2/internal/plugins/common/proxy/context.go:176]: task context is always canceled when collect is completed so fetch will not block forever. For a given context (related to task) context not changes during next Collect (it's just new context)

Regarding locking it's aligned to other context.Context related methods. Probably could return it without mock but wanted to leave it.

**Review by mpach-swi — APPROVED** [review]

**Review by mpach-swi — APPROVED** [review]

**Review by lootek — APPROVED** [review]

**Pull request merged**